pkg/controller/s3/bucket.go

```diff
@@ -168,11 +185,11 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 		case bucket.NeedsDeletion:
 			err = awsClient.Delete(ctx, cr)
 			if err != nil {
-				return managed.ExternalUpdate{}, awsclient.Wrap(err, errDelete)
+				return managed.ExternalUpdate{}, errors.Wrap(err, errDelete)
 			}
 		case bucket.NeedsUpdate:
 			if err := awsClient.CreateOrUpdate(ctx, cr); err != nil {
-				return managed.ExternalUpdate{}, awsclient.Wrap(err, errCreateOrUpdate)
+				return managed.ExternalUpdate{}, errors.Wrap(err, errCreateOrUpdate)
 			}
 		}
 	}
```

Comment:
Unrelated: managed reconciler already takes care of reconcile success condition, so you can remove line `196`.